app/models/user.rb

```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  has_many :posts
+  has_many :comments
+  has_many :likes
+
+  def recent_posts
+    Post.where(author_id: id).order(created_at: :desc).limit(3)
+  end
```

Comment:
- [ ] The `recent_posts` method uses `Post.where(author_id: id)` to retrieve the user's posts. Since you have already defined the association `has_many :posts` in the User model, you can use the association directly to fetch the recent posts. Here's an alternative implementation that leverages the association:
```ruby
def recent_posts
  posts.order(created_at: :desc).limit(3)
end
```

This alternative implementation simplifies the code by utilizing the association and provides the same result.

Reply:
Fixed.